python/cudf/cudf/core/series.py

```diff
@@ -2505,7 +2505,16 @@ def _align_to_index(
                 raise ValueError("Cannot align indices with non-unique values")
         lhs = self.to_frame(0)
         rhs = cudf.DataFrame(index=as_index(index))
-        result = lhs.join(rhs, how=how, sort=sort)[0]
+        if how == "left":
+            lhs["____"] = cupy.arange(len(lhs))
```

Comment:
Discussed offline and we're going to use `uuid4` here to ensure a unique column name